staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -156,3 +158,39 @@ func compileRule(rule apiextensions.ValidationRule, env *cel.Env) (compilationRe
 func generateUniqueSelfTypeName() string {
 	return fmt.Sprintf("selfType%d", time.Now().Nanosecond())
 }
+
+type celEstimator struct {
+	root *celmodel.DeclType
+}
+
+func (c *celEstimator) EstimateSize(element checker.AstNode) *checker.SizeEstimate {
+	currentNode := c.root
+	// cut off "self" from path, since we always start there
+	for _, name := range element.Path()[1:] {
```

Comment:
is Path() always 1 or more elements?

Reply:
handle nil?

> // Path returns a field path through the provided type declarations to the type of the AstNode, or nil if the AstNode does not
>	// represent type directly reachable from the provided type declarations.

Reply:
`Path()` does seem to return an empty array at times - though as the doc mentions, thankfully handling that corner cases is straightforward (likewise return `nil`, since if the type isn't reachable we can't provide a cost estimate for it).